proposals/P0900.bs

```diff
@@ -0,0 +1,184 @@
+the "value" of that property. Note that most properties (at least of those
+proposed thus far) will have a mode type that is merely a boolean, with their
+mode indicated by their presence or abscence in the `Properties...` parameter
+pack. Another example of a non-boolean modal property is the layout property
+proposed in [[P0009r4]].
+
+Convertibility
+--------------
+
+A follow-on to the orthogonality design decision is that of convertibility (via
+construction or assignment).  In similar interest of avoiding exponential
+explosion of pairwise convertibility specification and implementation, it makes
+sense to impose the constraint that an `mdspan` with *any* set of access
+properties is always convertible to an `mdspan` with any other set of access
+properties (and likewise for `span`).  A slight nuance that could be added to
+this constraint is to allow properties to forbid interconversion between
+different *modes* within the same property.  For instance, with the access mode
+property proposed in [[P0367r0]], if the `read` mode of that property is
+specified to mean that only reads will be done from the data in the `mdspan` or
+objects derived from it, then it is a clear violation of the contract to convert
+such an `mdspan` into one with the `write` mode of that property.  It is not
+unreasonable to allow this prohibition to be made within the type system, since
+the specification complexity is always linear in the number of modes of the
+property, and since the modes of a given property are not subject to the same
+extensibility constraints as properties themselves.
+
+
+On the Importance of API Invariance
+-----------------------------------
+
+In the design of many customization points analogous to the access properties
+proposed herein, invariance of the API for members of the resulting class
+template instantiation is cricitcal to its usefulness in generic context. For
+example, the invariance of the `vector` API with respect to the `Allocator`
+template parameter customization point enables the following generic code to be
+viable:
+
+```
+template <typename T, typename Allocator>
+auto copy_odd_values(const vector<T, Allocator>& v) {
+    vector<T, Allocator> ret;
+    auto is_odd = [](T const& i) { return i % 2 != 0; };
+    std::copy_if(v.begin(), v.end(), std::back_insert_iterator(ret), is_odd);
+    return ret;
+}
+```
+
+In the context of access properties for `span` and `mdspan`, this invariance is
+actually not quite as critical for several important reasons. Most importantly,
+both `span` and `mdspan` are intended to be passed by value, and thus the need
+to template on the `Properties...` template parameters to, e.g., avoid spurious
+copies to `const` reference parameters is not as acute. In fact, some access
+properties represent a contract that would be dangerous to transmit into a
+generic context, as is the case with `restrict_access` (proposed in P0856). In
```

Comment:
It might help to give an example of why this is dangerous (or point to one in e.g., P0856).